### PR TITLE
Platform error 500 when content-type is missing

### PR DESCRIPTION
**Pull request by kkarthik4cs**
Issue #SB-2868 fix: functional testcases for platform error(500) when content-type was missing

### CODE REVIEW

File: sunbird_service_api_test/src/test/java/org/sunbird/common/util/Constant.java [hunk 1]
```diff
@@ -19,8 +19,10 @@ private Constant() {}
   public static final String AUTHORIZATION = "Authorization";
   public static final String BEARER = "Bearer ";
   public static final String CONTENT_TYPE_APPLICATION_JSON = "application/json";
+  public static final String CONTENT_TYPE_APPLICATION_JSON_LD = "application/json-ld";
   public static final String EQUAL_SIGN = "=";
   public static final String USER_TEMPLATE_LOCATION = "templates/user/create/";
+  public static final String RESPONSE_TEMPLATE_LOCATION = "templates/responses/";
```
Comment (manzarul) [inline sunbird_service_api_test/src/test/java/org/sunbird/common/util/Constant.java]: change it templates/responses/ to  templates/malformed

File: sunbird_service_api_test/src/test/java/org/sunbird/integration/test/malformedRequest/MalformedRequestTest.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+import com.consol.citrus.annotations.CitrusTest;
+import com.consol.citrus.context.TestContext;
+import com.consol.citrus.dsl.testng.TestNGCitrusTestDesigner;
+import com.consol.citrus.http.client.HttpClient;
+import com.consol.citrus.testng.CitrusParameters;
+import com.consol.citrus.validation.json.JsonMappingValidationCallback;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class MalformedRequestTest extends TestNGCitrusTestDesigner {
```
Comment (manzarul) [inline sunbird_service_api_test/src/test/java/org/sunbird/integration/test/malformedRequest/MalformedRequestTest.java]: Define the class comment.

File: sunbird_service_api_test/src/test/java/org/sunbird/integration/test/malformedRequest/MalformedRequestTest.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+						Constant.RESPONSE_TEMPLATE_LOCATION + "content_type_mandatory_response.json", "createNote",
+						"/api/note/v1/create" } };
+	}
+
+	/**
+	 * Test for create request without content-type header.
+	 *
+	 * @param requestJson
+	 * @param responseJson
+	 * @param testName
```
Comment (manzarul) [inline sunbird_service_api_test/src/test/java/org/sunbird/integration/test/malformedRequest/MalformedRequestTest.java]: Method is taking four parm , so need to define all four. and add the description of each param.

File: sunbird_service_api_test/src/test/java/org/sunbird/integration/test/malformedRequest/MalformedRequestTest.java [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+	 */
+	@Test(dataProvider = "createRequestDataProvider", dependsOnMethods = { "getAdminAuthToken", })
```
Comment (manzarul) [inline sunbird_service_api_test/src/test/java/org/sunbird/integration/test/malformedRequest/MalformedRequestTest.java]: typo : "getAdminAuthToken", to "getAdminAuthToken"

File: sunbird_service_api_test/src/test/java/org/sunbird/integration/test/malformedRequest/MalformedRequestTest.java [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+	@CitrusParameters({ "requestJson", "responseJson", "testName", "url" })
+	@CitrusTest
+	public void createWithoutContentType(String requestJson, String responseJson, String testName, String url) {
```
Comment (manzarul) [inline sunbird_service_api_test/src/test/java/org/sunbird/integration/test/malformedRequest/MalformedRequestTest.java]: change createWithoutContentType to testRequestWithoutContentType

File: sunbird_service_api_test/src/test/java/org/sunbird/integration/test/malformedRequest/MalformedRequestTest.java [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+		getTestCase().setName(testName);
+		System.out.println("request: == " + requestJson);
```
Comment (manzarul) [inline sunbird_service_api_test/src/test/java/org/sunbird/integration/test/malformedRequest/MalformedRequestTest.java]: remove : System.out.println("request: == " + requestJson);

File: sunbird_service_api_test/src/test/java/org/sunbird/integration/test/malformedRequest/MalformedRequestTest.java [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+		http().client(restTestClient).receive().response(HttpStatus.BAD_REQUEST)
+				.payload(new ClassPathResource(responseJson));
+
+	}
+
+	/**
+	 * Test for create request with invalid(json-ld) content-type header.
+	 *
+	 * @param requestJson
+	 * @param responseJson
```
Comment (manzarul) [inline sunbird_service_api_test/src/test/java/org/sunbird/integration/test/malformedRequest/MalformedRequestTest.java]: add comment for all input param.

File: sunbird_service_api_test/src/test/java/org/sunbird/integration/test/malformedRequest/MalformedRequestTest.java [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+	 * @param testName
+	 */
+	@Test(dataProvider = "createRequestDataProvider", dependsOnMethods = { "getAdminAuthToken", })
```
Comment (manzarul) [inline sunbird_service_api_test/src/test/java/org/sunbird/integration/test/malformedRequest/MalformedRequestTest.java]: typo: "getAdminAuthToken", to "getAdminAuthToken"

File: sunbird_service_api_test/src/test/java/org/sunbird/integration/test/malformedRequest/MalformedRequestTest.java [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+	@CitrusParameters({ "requestJson", "responseJson", "testName", "url" })
+	@CitrusTest
+	public void createWithInvalidContentType(String requestJson, String responseJson, String testName, String url) {
```
Comment (manzarul) [inline sunbird_service_api_test/src/test/java/org/sunbird/integration/test/malformedRequest/MalformedRequestTest.java]: change createWithInvalidContentType to testRequestWithInvalidContentType

File: sunbird_service_api_test/src/test/java/org/sunbird/integration/test/malformedRequest/MalformedRequestTest.java [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+
+		http().client(restTestClient).receive().response(HttpStatus.BAD_REQUEST)
+				.payload(new ClassPathResource(responseJson));
+
+	}
+
+	@Test()
+	@CitrusTest
+	/**
+	 * Key cloak admin token generation is required , because on sunbird dev server
```
Comment (manzarul) [inline sunbird_service_api_test/src/test/java/org/sunbird/integration/test/malformedRequest/MalformedRequestTest.java]: Change the description  , it copy past from another method.

File: sunbird_service_api_test/src/test/java/org/sunbird/integration/test/malformedRequest/MalformedRequestTest.java [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+	 * after creating user , user have to login first then only his/her account will
+	 * be active. so we need to disable that option for created user only. That
+	 * option can be disable using keycloak admin auth token. So this method will
+	 * generate auth token and that token will be used in
+	 * **updateUserRequiredLoginActionTest** method.
+	 */
+	public void getAdminAuthToken() {
```
Comment (manzarul) [inline sunbird_service_api_test/src/test/java/org/sunbird/integration/test/malformedRequest/MalformedRequestTest.java]: Can we put it into common place , to reuse it.

Comment (kkarthik4cs) [reply sunbird_service_api_test/src/test/java/org/sunbird/integration/test/malformedRequest/MalformedRequestTest.java]: we can decide later whether can be moved into common baseTest for all testcases

**Pull request merged**